### PR TITLE
Removing some linker warnings that are no longer needed

### PR DESCRIPTION
**Pull request by joperezr**
cc: @eerhardt 

This will need to be rebased with https://github.com/dotnet/runtime/pull/52272 once that is merged, but sending this in the meantime to test against the CI platform matrix to ensure the other warnings aren't required any longer.

### CODE REVIEW

File: src/libraries/illink-sharedframework.targets [hunk 1]
```diff
@@ -14,19 +14,11 @@
...
-           IL2009: Could not find method A in type B specified in resource C
-           IL2025: Duplicate preserve of A in B
-      -->
-      <LinkerNoWarn>IL2009;IL2025</LinkerNoWarn>
       <!-- https://github.com/dotnet/runtime/issues/40336 - need to also suppress the following on non-Windows:
            IL2008: Could not find type A specified in resource B
-           and on 32-bit builds:
-           IL2012: Could not find field 'Padding' in type 'System.Runtime.CompilerServices.RawArrayData'
       -->
       <LinkerNoWarn Condition="'$(TargetOS)' != 'windows'">$(LinkerNoWarn);IL2008</LinkerNoWarn>
```
Comment (eerhardt) [inline src/libraries/illink-sharedframework.targets:20]:
```suggestion
      <LinkerNoWarn Condition="'$(TargetOS)' != 'windows'">IL2008</LinkerNoWarn>
```

This is the place that defines this property (see line 21 previously). No need to append a `;` to the beginning of the list :)

**Review by eerhardt — APPROVED** [review]
This is awesome we can remove this many warning suppressions! Let's merge this and delete code.

Separately, we should see what we can do to remove that last `IL2008` suppression.

cc @vitek-karas @sbomer

File: src/coreclr/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets [hunk 2]
```diff
@@ -14,7 +14,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <_ILLinkDescriptorsFilePaths Include="$(ILLinkDirectory)ILLink.Descriptors.xml" />
+    <_ILLinkDescriptorsFilePaths Include="$(ILLinkDirectory)ILLink.Descriptors.Common.xml" />
```
Comment (eerhardt) [inline src/coreclr/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets:17]: I'd call this `Shared.xml`, we use that term elsewhere. e.g.

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml

Comment (eerhardt) [reply src/coreclr/System.Private.CoreLib/CreateRuntimeRootILLinkDescriptorFile.targets:17]: Or... maybe just leave it `ILLink.Descriptors.xml`, and also have `ILLink.Descriptors.Windows.xml`.

**Comment by vitek-karas** [comment]
If this completely removes all warnings from non-Windows builds we should resolve https://github.com/dotnet/runtime/issues/40336 and also revert the changes made in https://github.com/dotnet/sdk/pull/16979. That work is at least partially tracked by https://github.com/mono/linker/issues/1973.

@sbomer

File: src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.NonBrowser.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<linker>
```
Comment (marek-safar) [inline src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.NonBrowser.xml:1]: Should be called something like `.EventSource.xml` instead

**Review by marek-safar — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.NonBrowser.xml (hunk 3 above)
Comment (joperezr) [reply src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.NonBrowser.xml:1]: I was going for having the name of the file to indicate build configuration inclusion as opposed to actual contents of it, which is what we've done in the rest of the repo for similar file exclusions/inclusions. That said, if you feel strongly about the name being wrong then I can rename it.

Comment (marek-safar) [reply src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.NonBrowser.xml:1]: I don't feel strongly about this but mixing features and platforms is always problematic

**Pull request merged**